src/lib.rs

```diff
@@ -62,165 +69,208 @@ use self::consts::*;
 #[derive(Debug)]
 pub struct WrongClock;
 
-/// Ethernet driver for *STM32* chips.
-pub struct Eth<'rx, 'tx> {
-    eth_mac: ETHERNET_MAC,
+/// Ethernet DMA.
+pub struct EthernetDMA<'rx, 'tx> {
     eth_dma: ETHERNET_DMA,
     rx_ring: RxRing<'rx>,
     tx_ring: TxRing<'tx>,
 }
+/// Ethernet media access control (MAC).
+pub struct EthernetMAC {
+    eth_mac: ETHERNET_MAC,
+}
 
-impl<'rx, 'tx> Eth<'rx, 'tx> {
-    /// Initialize and start tx and rx DMA engines.
-    ///
-    /// Make sure that the buffers reside in a memory region that is
-    /// accessible by the peripheral. Core-Coupled Memory (CCM) is
-    /// usually not accessible. HCLK must be at least 25MHz.
-    ///
-    /// Uses an interrupt free critical section to turn on the ethernet clock
-    /// for STM32F7xx.
-    ///
-    /// Other than that, initializes and starts the Ethernet hardware
-    /// so that you can [`send()`](#method.send) and
-    /// [`recv_next()`](#method.recv_next).
-    pub fn new<REFCLK, CRS, TXEN, TXD0, TXD1, RXD0, RXD1>(
-        eth_mac: ETHERNET_MAC,
-        eth_dma: ETHERNET_DMA,
-        rx_buffer: &'rx mut [RxRingEntry],
-        tx_buffer: &'tx mut [TxRingEntry],
-        clocks: Clocks,
-        pins: EthPins<REFCLK, CRS, TXEN, TXD0, TXD1, RXD0, RXD1>,
-    ) -> Result<Self, WrongClock>
-    where
-        REFCLK: RmiiRefClk + AlternateVeryHighSpeed,
-        CRS: RmiiCrsDv + AlternateVeryHighSpeed,
-        TXEN: RmiiTxEN + AlternateVeryHighSpeed,
-        TXD0: RmiiTxD0 + AlternateVeryHighSpeed,
-        TXD1: RmiiTxD1 + AlternateVeryHighSpeed,
-        RXD0: RmiiRxD0 + AlternateVeryHighSpeed,
-        RXD1: RmiiRxD1 + AlternateVeryHighSpeed,
-    {
-        setup::setup();
-        pins.setup_pins();
-        let mut eth = Eth {
-            eth_mac,
-            eth_dma,
-            rx_ring: RxRing::new(rx_buffer),
-            tx_ring: TxRing::new(tx_buffer),
-        };
-        eth.init(clocks)?;
-        eth.rx_ring.start(&eth.eth_dma);
-        eth.tx_ring.start(&eth.eth_dma);
-        Ok(eth)
-    }
+/// Create and initialise the ethernet driver.
+///
+/// Initialize and start tx and rx DMA engines.
+/// Sets up the peripheral clocks and GPIO configuration,
+/// and configures the ETH MAC and DMA peripherals.
+/// Automatically sets slew rate to VeryHigh.
```

Comment:
This is not true for the F1 code.

Reply:
Which part?

Reply:
"Automatically sets slew rate to very high"

Reply:
The slew rate is now automatically set to `VeryHigh` (or `MHz50`) for `stm32f107` as well